.github/workflows/update-wpt.yml

```diff
@@ -0,0 +1,46 @@
+name: Update WPT
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: 0 0 * * *
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  update-wpt:
+    name: Update WPT
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633  # v4.1.2
+      - name: Update WPT
+        run: |
+          rm -rf test/fixtures/wpt && mkdir test/fixtures/wpt &&
+
+          git clone https://github.com/web-platform-tests/wpt.git --depth=1 test/fixtures/tmp-wpt &&
```

Comment:
You could replace this with an extra call to `actions/checkout`?

And sparse checkout only the folders we need directly into the folder we want them to go?